luigi/worker.py

```diff
@@ -180,16 +180,12 @@ def run(self):
             t0 = time.time()
             status = None
 
-            if _is_external(self.task):
-                # External task
-                # TODO(erikbern): We should check for task completeness after non-external tasks too!
-                # This will resolve #814 and make things a lot more consistent
```

Comment:
So does this fix this TODO?

Reply:
Yes if I got the intent right. @erikbern may need to take a look and see if I'm missing anything.